pkg/cosign/bundle/sign.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2025 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/sigstore/sigstore-go/pkg/root"
+	"github.com/sigstore/sigstore-go/pkg/sign"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"google.golang.org/protobuf/encoding/protojson"
+)
+
+func SignData(content sign.Content, keypair sign.Keypair, idToken string, signingConfig *root.SigningConfig, trustedMaterial root.TrustedMaterial) ([]byte, error) {
+	var opts sign.BundleOptions
+
+	if trustedMaterial != nil {
+		opts.TrustedRoot = trustedMaterial
+	}
+
+	if idToken != "" {
```

Comment:
Could fetching the idToken be moved here, instead of duplicating in sign_blob.go and attest_blob.go? This is already where the rest of the SelectService logic seems to be happening.

Reply:
Refactored identity token retrieval into its own method that's now called from sign_blob and attest_blob. When we move this out into a shared library and work towards v3, I'll want to refactor this a bit more - for example, providing an identity token via flag that's either the token itself or a path isn't the cleanest UX.